visualization/app/codeCharta/util/fileDownloader.ts

```diff
@@ -1,42 +1,50 @@
 import angular from "angular"
 import * as d3 from "d3"
-import { CCFile, CodeMapNode, BlacklistType, BlacklistItem, FileSettings, ExportCCFile, AttributeTypes } from "../codeCharta.model"
+import { CodeMapNode, BlacklistType, BlacklistItem, FileSettings, ExportCCFile, FileMeta, AttributeTypes } from "../codeCharta.model"
 import { DownloadCheckboxNames } from "../ui/dialog/dialog.download.component"
 import { CodeChartaService } from "../codeCharta.service"
 
 export class FileDownloader {
-	public static downloadCurrentMap(file: CCFile, downloadSettingsNames: string[], fileName: string) {
-		const exportCCFile: ExportCCFile = this.getProjectDataAsCCJsonFormat(file, downloadSettingsNames)
+	public static downloadCurrentMap(
+		map: CodeMapNode,
+		fileMeta: FileMeta,
+		s: FileSettings,
+		downloadSettingsNames: string[],
+		fileName: string
+	) {
+		const exportCCFile: ExportCCFile = this.getProjectDataAsCCJsonFormat(map, fileMeta, s, downloadSettingsNames)
 		const newFileNameWithExtension: string = fileName + CodeChartaService.CC_FILE_EXTENSION
 		this.downloadData(exportCCFile, newFileNameWithExtension)
 	}
 
-	private static getProjectDataAsCCJsonFormat(file: CCFile, downloadSettingsNames: string[]) {
-		const s: FileSettings = file.settings.fileSettings
-
-		let downloadObject: ExportCCFile = {
-			projectName: file.fileMeta.projectName,
-			apiVersion: file.fileMeta.apiVersion,
-			nodes: [this.removeJsonHashkeysAndVisibleAttribute(file.map)],
+	private static getProjectDataAsCCJsonFormat(
+		map: CodeMapNode,
+		fileMeta: FileMeta,
+		s: FileSettings,
```

Comment:
I'm not really a fan of single-letter variables. If you create a temporary variable, it's fine. For example `const s = myParameterString.subString()`
It's easier to understand when they're actually called `fileSettings : FileSettings`, because we often refer to settings as a temporary variable `s` as well

Reply:
Fair enough 👍 makes sense